Tests/TuistSupportTests/Utils/GlobTests.swift

```diff
@@ -0,0 +1,317 @@
+import Foundation
+import XCTest
+@testable import TuistSupport
+@testable import TuistSupportTesting
+
+//  Inspired by: https://gist.github.com/efirestone/ce01ae109e08772647eb061b3bb387c3
+
+final class GlobTests: TuistTestCase {
+    let tmpFiles = ["foo", "bar", "baz", "dir1/file1.ext", "dir1/dir2/dir3/file2.ext", "dir1/**(_:_:)/file3.ext"]
+    var tmpDir: URL!
```

Comment:
```suggestion
    private var tmpDir: AbsolutePath!
```

Let's try to use the Tuist primitives as much as possible